### ISSUE TITLE
Multiple remote subscribes result in duplicate callback evals

### ISSUE THREAD

**Issue by jonathanvanschenck** [Daxbot/entangld#6]
If a datastore subscribes to a remote datastore multiple times (same path, multiple callbacks), then when the remote store calls `.set` it results in both local callback functions getting called twice. Here's an example:

```js
let parent=new Entangld();
let child=new Entangld();


parent.attach("child",child);

parent.transmit((msg, store)=>store.receive(msg,parent));
child.transmit((msg, store)=>store.receive(msg, child));

// Attach first subscribe:
var call_count_1 = 0;
parent.subscribe('child.value', (path,value) => { call_count_1 += 1; console.log(`Sub 1 called for ${call_count_1}th time`) })
var call_count_2 = 0;
parent.subscribe('child.value', (path,value) => { call_count_2 += 1; console.log(`Sub 2 called for ${call_count_2}th time`) })

setTimeout(() => {
  child.set("value",0)
}, 50);

```

The output in the current version is:

```
Sub 1 called for 1th time
Sub 2 called for 1th time
Sub 1 called for 2th time
Sub 2 called for 2th time
```

so that both callbacks are called twice. I think that the correct functionality should either be:
 1) Both subscribes get called once:
```
Sub 1 called for 1th time
Sub 2 called for 1th time
```

  2) Only the last subscribe gets called once:
```
Sub 2 called for 1th time
```

The former seems more consistent wit the rest of the package.

**Comment by jonathanvanschenck** [comment]
See PR #7

**Comment by jonathanvanschenck** [comment]
Solved (mostly) by PR #9, due to the requirements for backwards compatibility, this issue doesn't get solved if the remote datastore isn't on the new version, but there isn't really a way around that.
